host/4/bookworm/python/python312/python312-appservice.Dockerfile

```diff
@@ -0,0 +1,108 @@
+    unzip /$EXTENSION_BUNDLE_FILENAME_V4 -d /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION_V4 && \
+    rm -f /$EXTENSION_BUNDLE_FILENAME_V4 &&\
+    find /FuncExtensionBundles/ -type f -exec chmod 644 {} \;
+
+FROM mcr.microsoft.com/oryx/python:3.12-debian-bookworm AS python
+FROM mcr.microsoft.com/dotnet/aspnet:8.0-bookworm-slim-amd64
+ARG HOST_VERSION
+    
+COPY --from=runtime-image ["/azure-functions-host", "/azure-functions-host"]
+COPY --from=runtime-image [ "/FuncExtensionBundles", "/FuncExtensionBundles" ]
+COPY install_ca_certificates.sh /opt/startup/
+COPY sshd_config /etc/ssh/
+COPY start.sh /azure-functions-host/
+RUN chmod +x /opt/startup/install_ca_certificates.sh && \
+    chmod +x /azure-functions-host/start.sh
+
+# Install Python dependencies
+RUN apt-get update && \
+    apt-get install -y wget apt-transport-https curl gnupg2 locales && \
+    curl -fsSL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor -o /usr/share/keyrings/microsoft-prod.gpg && \
+    echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections && \
+    echo "deb [arch=amd64] https://packages.microsoft.com/debian/12/prod bookworm main" | tee /etc/apt/sources.list.d/mssql-release.list && \
+    # Needed for libss3 and in turn MS SQL
+    echo 'deb http://security.debian.org/debian-security bookworm-security main' >> /etc/apt/sources.list && \
+    curl https://packages.microsoft.com/config/debian/12/prod.list | tee /etc/apt/sources.list.d/mssql-release.list && \
+    # install MS SQL related packages.pinned version in PR # 1012.
+    echo 'en_US.UTF-8 UTF-8' > /etc/locale.gen && \
+    locale-gen && \
+    apt-get update && \
+    # MS SQL related packages: unixodbc msodbcsql18 mssql-tools
+    ACCEPT_EULA=Y apt-get install -y unixodbc msodbcsql18 mssql-tools18 && \
+    # OpenCV dependencies:libglib2.0-0 libsm6 libxext6 libxrender-dev xvfb
+    apt-get install -y libglib2.0-0 libsm6 libxext6 libxrender-dev xvfb && \
+    # .NET Core dependencies: ca-certificates libc6 libgcc1 libgssapi-krb5-2 libicu72 libssl3 libstdc++6 zlib1g 
+    # Azure ML dependencies: liblttng-ust0
+    # OpenMP dependencies: libgomp1
+    # binutils: binutils
+    apt-get install -y --no-install-recommends ca-certificates \
+    libc6 libgcc1 libgssapi-krb5-2 libicu72 libssl3 libstdc++6 zlib1g && \
+    apt-get install -y libglib2.0-0 libsm6 libxext6 libxrender-dev xvfb binutils \
+    libgomp1 liblttng-ust1 && \
+    rm -rf /var/lib/apt/lists/* 
+
+COPY --from=runtime-image [ "/workers/python/3.12/LINUX", "/azure-functions-host/workers/python/3.12/LINUX" ]
+COPY --from=runtime-image [ "/workers/python/worker.config.json", "/azure-functions-host/workers/python" ]
+COPY --from=python [ "/opt", "/opt" ]
+
+# Link all binaries from /opt/python/3.12/bin to /usr/bin/
+RUN for file in /opt/python/3.12/bin/*; do \
```

Comment:
Is this still necessary? with the indicution of the LD lib I dont think we should sym link if we can avoid it

Reply:
We still need this. The LD commands links the lib dir. We would still need all the python binaires

Reply:
This set of commands is symbolicly linking all the files in python dir to files in the user/binary directory.

The LD commands tell linux to add the directory to the path. We should not need both

Reply:
the LD command links the shared libs. Without the linking, Pip does not work.